src/function_transformer_attention.py

```diff
@@ -186,13 +186,13 @@ def forward(self, x, edge):
       src = q[edge[0, :], :, :]
       dst_k = k[edge[1, :], :, :]
 
-    if self.opt['attention_type'] == "scaled_dot":
+    if not self.opt['beltrami'] and self.opt['attention_type'] == "exp_kernel":
+      prods = (torch.sum((src - dst_k) ** 2, dim=1) / (2 * self.lengthscale ** 2))
+    elif self.opt['attention_type'] == "scaled_dot":
       prods = torch.sum(src * dst_k, dim=1) / np.sqrt(self.d_k)
```

Comment:
this should be:
prods = self.output_var ** 2 * torch.exp(-(torch.sum((src - dst_k) ** 2, dim=1) / (2 * self.lengthscale ** 2)))

will also need to make changes to init, I'll fix after this PR